admin/server/organizations.go

```diff
@@ -527,6 +525,24 @@ func (s *Server) CreateAutoinviteDomain(ctx context.Context, req *adminv1.Create
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
+	if !claims.Superuser(ctx) {
+		if !claims.OrganizationPermissions(ctx, org.ID).ManageOrg {
+			return nil, status.Error(codes.PermissionDenied, "only org admins can add whitelisted domain")
+		}
+		// check if the user's domain matches the autoinvite domain
+		user, err := s.admin.DB.FindUser(ctx, claims.OwnerID())
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+		if !strings.HasSuffix(user.Email, req.Domain) {
+			return nil, status.Error(codes.PermissionDenied, "Domain name doesn’t match verified email domain. Please contact Rill support.")
+		}
```

Comment:
Probably need to parse email and check. This way, I could sign up as `benjamin@notfoo.com` and whitelist `foo.com`

Reply:
changing to `strings.HasSuffix(user.Email, "@" + req.Domain)`